hist/hist/src/TH1.cxx

```diff
@@ -8230,11 +8247,17 @@ Double_t TH1::KolmogorovTest(const TH1 *h2, Option_t *option) const
 
       // make nEXPT experiments (this should be a parameter)
       prb3 = 0;
+      if (afunc1) Copy(*h1Expt);
+      if (afunc2) h2->Copy(*h2Expt);
       for (Int_t i=0; i < nEXPT; i++) {
-         h1Expt->Reset();
-         h2Expt->Reset();
-         h1Expt->FillRandom(h1_cpy, (Int_t)esum1);
-         h2Expt->FillRandom(h1_cpy, (Int_t)esum2);
+         if (!afunc1) {
+            h1Expt->Reset();
+            h1Expt->FillRandom(h1_cpy, (Int_t)esum1);
+         }
+         if (!afunc2) {
+            h2Expt->Reset();
+            h2Expt->FillRandom(h1_cpy, (Int_t)esum2);
```

Comment:
Why is `h2Expt` a randomized version of `h1_cpy` here? Should it not be drawn from h2?

Reply:
No, here we need to find the distribution of the KS test statistics, dsEXPT for the NULL, i.e. when we draw two histograms from the same parent distribution. 
However, you are right there is an error. If afunc1=TRUE we need to draw from h2. So the logic is not correct.